qlib/kernel/asm/aarch64.rs

```diff
@@ -639,3 +639,28 @@ pub fn spsel() -> u64 {
     }
     ret
 }
+
+// Setting or clearing the (hardware) privilege access never bit. If set, the
+// CPU will be trapped when trying to access EL0(user) memory from EL1 or above.
+// This is to prevent the kernel from accessing user memory accidentally,
+#[cfg(target_arch = "aarch64")]
+#[inline]
+pub fn SetPAN() {
+    unsafe {
+        asm!(
+            "msr PAN, #1",
```

Comment:
why does ClearPAN need the memory barrier, and SetPAN doesn't?

Reply:
on a second thought the fences should not be here, sorry.

I'll delete